amplitude.js

```diff
@@ -801,18 +830,41 @@ AmplitudeClient.prototype._setInStorage = function _setInStorage(storage, key, v
 };
 
 /**
+ * Helper function to remove values in storage
+ * Storage argument allows for localStoraoge and sessionStoraoge
+ * @private
+ */
+AmplitudeClient.prototype._removeFromStorage = function _removeFromStorage(storage, key) {
+  storage.removeItem(key + this._storageSuffix);
+};
+
+/**
  * cookieData (deviceId, userId, optOut, sessionId, lastEventTime, eventId, identifyId, sequenceNumber)
  * can be stored in many different places (localStorage, cookie, etc).
  * Need to unify all sources into one place with a one-time upgrade/migration.
  * @private
  */
 var _upgradeCookeData = function _upgradeCookeData(scope) {
-  // skip if migration already happened
-  var cookieData = scope.cookieStorage.get(scope.options.cookieName);
+  // check if already migrated to new instance keys with scope
+  var scopedKey = scope.options.cookieName + scope._apiKeySuffix + scope._storageSuffix;
```

Comment:
Oh my bad, you're right this is in cookie, and cookies have a max 4093 limit. I was thinking about compressing the key, like slicing x characters from it, but say we slice like 10, then we're only saving like 22 characters. If that's enough to make a difference then the website is already past like 4000 bytes used out of 4093, and they might have other issues to worry about.

Another thing I just remembered is that our cookie storage wrapper will fall back to localstorage if cookies aren't available. When the sdk initializes, it will attempt to write a test cookie with a test value, and then delete it after. If it fails due to full cookie usage, then it will fall back to localstorage. The only issue in that case is the lack of cross-subdomain support, but I think we've covered like 99% of our users